app/helpers/photo_helper.rb

```diff
@@ -0,0 +1,88 @@
+module PhotoHelper
+
+  def resize_photo photo, params
+    rover_name = photo.rover.name.downcase
+    suffix_data = lookup_suffix rover_name, params[:size]
+
+    if !suffix_data.nil?
+      replace_photo_suffix photo, suffix_data[:old_length], suffix_data[:new]
+    else
+      'size error'
```

Comment:
Rather than return a string here, it would probably be better to `raise` an error. You could create a customer error class, perhaps called `InvalidSizeParameter` or something similar. Then in the controller, you can do something like
```
rescue StandardError => e
  render json: { errors: e.full_message }, status: :bad_request
end
```
I would even maybe refactor the invalid rover name to work the same way.

Reply:
Thank you for the feedback! I put something together extending StandardError. Let me know what you think. It is my first time raising an error in Ruby